src/libraries/System.Net.Security/tests/FunctionalTests/TestHelper.cs

```diff
@@ -47,5 +48,137 @@ internal static (VirtualNetworkStream ClientStream, VirtualNetworkStream ServerS
+                    new X509SubjectKeyIdentifierExtension(rootReq.PublicKey, false));
+                rootReq.CertificateExtensions.Add(
+                    new X509KeyUsageExtension(X509KeyUsageFlags.KeyCertSign, false));
+
+                //DateTimeOffset start = DateTimeOffset.UtcNow.AddMinutes(-5);
+                DateTimeOffset endTime = startTime.AddMonths(1);
+
+                X509Certificate2 rootCertWithKey = rootReq.CreateSelfSigned(startTime, endTime);
+
+                CertificateRequest intermedReq = new CertificateRequest(
+                   "CN=Intermediate",
+                   intermediate,
+                   HashAlgorithmName.SHA256,
+                   RSASignaturePadding.Pkcs1);
+
+                intermedReq.CertificateExtensions.Add(
+                    new X509BasicConstraintsExtension(true, false, 0, true));
+                intermedReq.CertificateExtensions.Add(
+                    new X509SubjectKeyIdentifierExtension(intermedReq.PublicKey, false));
+                intermedReq.CertificateExtensions.Add(
+                    new X509KeyUsageExtension(X509KeyUsageFlags.KeyCertSign, false));
+
+                byte[] serial = new byte[8];
+                RandomNumberGenerator.Fill(serial);
+
+                X509Certificate2 intermedCertWithKey;
+                using (X509Certificate2 intermedPub = intermedReq.Create(rootCertWithKey, startTime, endTime, serial))
+                {
+                    intermedCertWithKey = intermedPub.CopyWithPrivateKey(intermediate);
+                }
+
+                CertificateRequest serverReq = new CertificateRequest(
+                     $"CN={name}",
+                     server,
+                     HashAlgorithmName.SHA256,
+                     RSASignaturePadding.Pkcs1);
+
+                serverReq.CertificateExtensions.Add(
+                    new X509BasicConstraintsExtension(false, false, 0, false));
+                serverReq.CertificateExtensions.Add(
+                    new X509SubjectKeyIdentifierExtension(serverReq.PublicKey, false));
+
+                // Add Issuer KeyIdentifier
+                using (SHA1 sha1 = SHA1.Create())
+                {
+                    byte[] data = new byte[24];
+                    data[0] = 0x30; //SEQUENCE
+                    data[1] = 22;
+                    data[2] = 0x80;
+                    data[3] = 20;
+                    Buffer.BlockCopy(sha1.ComputeHash(intermedCertWithKey.PublicKey.EncodedKeyValue.RawData), 0, data, 4, 20);
+                    serverReq.CertificateExtensions.Add(new X509Extension(new Oid("2.5.29.35"), data, false));
+                }
+
+                // 1.3.6.1.5.5.7.1.1
+                if (caUrl != null)
+                {
+                    var urlBytes = Encoding.ASCII.GetBytes(caUrl);
+
+                    byte[] data = new byte[urlBytes.Length + 16];
+                    data[0] = 0x30; //SEQUENCE
+                    data[1] = (byte)(urlBytes.Length + 14);
```

Comment:
This whole thing only works if `urlBytes.Length + 14 <= 0x7F`.  Might want an assert, or to just use AsnWriter, or perhaps move my test CertificateAuthority into shared test source :)